base/inference.jl

```diff
@@ -2415,6 +2416,7 @@ function inlineable(f::ANY, e::Expr, atype::ANY, sv::StaticVarInfo, enclosing_as
 
     body = Expr(:block)
     body.args = ast.args[3].args::Array{Any,1}
+    propagate_inbounds, _ = popmeta!(body, :propagate_inbounds)
```

Comment:
I don't know, but I would guess that most of `inference.jl` was written at a time when that wasn't available at this point in the bootstrap. If it's working for you, maybe now that functionality is available (perhaps due to @vtjnash's reorganization in #11274)?

This is pure speculation on my part.